docs/source/scaleVM.md

```diff
@@ -0,0 +1,165 @@
+# 🖥️ Scale Fedora & Windows VMs with Benchmark Runner
+
+This document provides automated scripts and configuration to **scale Fedora and Windows VMs** on OpenShift clusters. It verifies the VM login page and measures **bootstorm time**—the time it takes for VMs to become fully operational.
+
+---
+
+## 🚀 Features
+
+- Scale Fedora 37 VMs using **ephemeral storage**
+- Scale Windows 11 VMs using **ODF-backed storage** (via DataVolume)
+- Parallel VM loading across specified worker nodes
+- Optional metrics collection via **Elasticsearch**
+- Logs and artifacts are stored locally for post-run analysis
+
+---
+
+## 🧪 Prerequisites
+
+- OpenShift cluster with access to worker nodes
+- [`podman`](https://podman.io) installed
+- Access to the [benchmark-runner](https://quay.io/repository/ebattat/benchmark-runner) container image
+- Properly configured `KUBECONFIG` for cluster access
+
+---
+
+## ⚙️ Parameters
+
+### Common Parameters
+| Variable | Description |
+|----------|-------------|
+| `KUBEADMIN_PASSWORD` | OpenShift admin password |
+| `KUBECONFIG_PATH` | Path to your kubeconfig (e.g. `/root/.kube/config`) |
+| `SCALE_NODES` | List of target worker nodes (e.g. `"[ 'worker-0', 'worker-1' ]"`) |
+| `THREADS_LIMIT` | Number of physical CPUs to use for parallelism |
+| `TIMEOUT` | Maximum duration (in seconds) for the run |
+| `RUN_ARTIFACTS_URL` | (Optional) Remote URL for saving logs |
+| `SAVE_ARTIFACTS_LOCAL` | If true, saves logs locally to `/tmp/benchmark-runner-run-artifacts` |
+
+---
+
+## 🧰 Fedora VM BootStorm (Ephemeral Storage)
+
+This mode loads Fedora VMs using ephemeral storage and verifies their readiness via the login page.
+
+### Fedora-Specific Parameters
+
+| Variable | Description                                                |
+|----------|------------------------------------------------------------|
+| `BOOTSTORM_SCALE` | Number of Fedora VMs per worker node                       |
+| `FEDORA_CONTAINER_DISK` | (Optional) Fedora .qcow2 image from quay.io (default: Fedora 37) |
+
+### Example Script
+
+```bash
+#!/bin/bash
+
+# Load Fedora 37 VMs including verification (verify login page)
+sudo podman run --rm -it \
+  -e WORKLOAD='bootstorm_vm' \
+  -e KUBEADMIN_PASSWORD="${KUBEADMIN_PASSWORD}" \
+  -e RUN_ARTIFACTS_URL="${RUN_ARTIFACTS_URL}" \
+  -e SAVE_ARTIFACTS_LOCAL=True \
+  -e SCALE="${BOOTSTORM_SCALE}" \
+  -e SCALE_NODES="${SCALE_NODES}" \
+  -e THREADS_LIMIT="${THREADS_LIMIT}" \
+  -e RUN_STRATEGY=True \
+  -e TIMEOUT="${TIMEOUT}" \
+  -e log_level="INFO" \
+  -v "${KUBECONFIG_PATH}:${KUBECONFIG_PATH}" \
+  -v /tmp/benchmark-runner-run-artifacts:/tmp/benchmark-runner-run-artifacts \
+  --privileged quay.io/ebattat/benchmark-runner:latest
+```
+
+## 🧰 Windows VM Bootstorm (ODF Storage)
+
+This mode loads Windows VMs using ODF persistent volumes and validates successful login.
```

Comment:
It should (probably) also be possible to do this with ephemeral storage.

Reply:
Currently its support only ODF storage, we can add ephemeral support in the future.

Reply:
Yes, see above.